gradio/external.py

```diff
@@ -647,6 +797,23 @@ def open_api_stream(
                 response += chunk.choices[0].delta.content
                 yield response
 
+    supported_extensions = []
+    for file_type in file_types:
+        if file_type == "text_encoded":
+            supported_extensions += TEXT_FILE_EXTENSIONS
+        elif file_type == "image":
+            supported_extensions += IMAGE_FILE_EXTENSIONS
+        else:
+            raise ValueError(
+                f"Invalid file type: {file_type}. Must be 'text_encoded' or 'image'."
+            )
+
     return ChatInterface(
-        open_api_stream if streaming else open_api, type="messages", **kwargs
+        open_api_stream if streaming else open_api,
+        type="messages",
+        multimodal=bool(file_types),
+        textbox=gradio.MultimodalTextbox(file_types=supported_extensions)
+        if file_types
+        else None,
```

Comment:
Just fyi this only allows uploading a single image at a time, whereas the api can support multiple changes I believe. To change, set `file_count="multiple"` in `gr.MultimodalTextbox`